girder/utility/mail_utils.py

```diff
@@ -117,15 +117,48 @@ def addTemplateDirectory(dir, prepend=False):
     _templateLookup.directories.insert(idx, dir)
 
 
+class _SMTPConnection(object):
+    def __init__(self, host, port=None, encryption=None,
+                 username=None, password=None):
+        self.host = host
+        self.port = port
+        self.encryption = encryption
+        self.username = username
+        self.password = password
+
+    def __enter__(self):
+        if self.encryption == 'ssl':
+            self.connection = smtplib.SMTP_SSL(self.host, self.port)
+        else:
+            self.connection = smtplib.SMTP(self.host, self.port)
+            if self.encryption == 'starttls':
+                self.connection.starttls()
+        if self.username and self.password:
+            self.connection.login(self.username, self.password)
+
+    def send(self, fromAddress, toAddresses, message):
+        self.connection.sendmail(fromAddress, toAddresses, message)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.quit()
+
```

Comment:
I think we should make this in the public API and make `__enter__` return `self`.